spec/integration/autobahn_intg_spec.rb

```diff
@@ -319,6 +319,20 @@ def create_transport_system(name, options={})
         await_delivery
         @queues.map { |q| q.status.first }.reduce(:+).should == 1
       end
+
+      it 'does not deliver any new messages after it has been unsubscribed' do
+        unsubscribed = Autobahn::Concurrency::Semaphore.new(0)
+        @consumer.subscribe do |headers, message|
+          begin
+            unsubscribed.acquire
+            headers.ack
+          rescue java.lang.InterruptedException
+          end
+        end
+        unsubscribed.release
+        @consumer.unsubscribe!
```

Comment:
To clarify, the expectation `unsubscribed.has_queued_threads.should be_false` might be correct in both cases, but it seems random which order the methods are called in the test